File: frontend/src/types/email.ts
export interface Email {
  id: string
  message_id: string
  sender_email: string
  sender_name: string | null
  reply_to: string | null
  recipient_email: string
  recipients_cc: string[]
  subject: string | null
  body_text: string | null
  body_html: string | null
  headers: Record<string, unknown>
  urls: unknown[]
  attachments: unknown[]
  auth_results: Record<string, unknown>
  received_at: string | null
  ingested_at: string
  created_at: string
  risk_level: string | null
  verdict: string | null
  final_score: number | null
}

export interface EmailList {
  items: Email[]
  total: number
  page: number
  size: number
}

export interface EmailIngest {
  message_id: string
  sender_email: string
  sender_name?: string
  reply_to?: string
  recipient_email: string
  recipients_cc?: string[]
  subject?: string
  body_text?: string
  body_html?: string
  headers?: Record<string, unknown>
  urls?: unknown[]
  attachments?: unknown[]
  auth_results?: Record<string, unknown>
  received_at?: string
}

File: backend/app/schemas/email.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class EmailIngest(BaseModel):
    message_id: str
    sender_email: str
    sender_name: str | None = None
    reply_to: str | None = None
    recipient_email: str
    recipients_cc: list[str] = []
    subject: str | None = None
    body_text: str | None = None
    body_html: str | None = None
    headers: dict = {}
    urls: list = []
    attachments: list = []
    auth_results: dict = {}
    received_at: datetime | None = None


class EmailResponse(BaseModel):
    id: UUID
    message_id: str
    sender_email: str
    sender_name: str | None
    reply_to: str | None
    recipient_email: str
    recipients_cc: list[str]
    subject: str | None
    body_text: str | None
    headers: dict
    urls: list
    attachments: list
    auth_results: dict
    received_at: datetime | None
    ingested_at: datetime
    created_at: datetime

    model_config = {"from_attributes": True}


class EmailListItemResponse(EmailResponse):
    risk_level: str | None = None
    verdict: str | None = None
    final_score: float | None = None


class EmailList(BaseModel):
    items: list[EmailListItemResponse]
    total: int
    page: int
    size: int
